app/forms/two_factor_options_form.rb

```diff
@@ -27,7 +27,7 @@ def submit(params)
   private
 
   def validate_selection_present
-    return if !has_no_mfa_or_in_required_flow? || selection.present?
+    return if selection.present? || has_minimum_required_mfa_methods?
```

Comment:
I kinda snuck the rearranging of the `selection.present?` check into the diff in https://github.com/18F/identity-idp/pull/9119#discussion_r1313302517, but the nice side-effect is that it should prevent a potential handful of database queries for the majority case where a user makes a selection before submitting the form, thanks to short-circuiting logic. 🎉